// @chakra-ui/cli/dist/utils/init-cli.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.initCLI = initCLI;

var _cliCheckNode = _interopRequireDefault(require("cli-check-node"));

var _cliWelcome = _interopRequireDefault(require("cli-welcome"));

var _updateNotifier = _interopRequireDefault(require("update-notifier"));

var _cliHandleUnhandled = _interopRequireDefault(require("cli-handle-unhandled"));

var _package = _interopRequireDefault(require("../../package.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

async function initCLI() {
  (0, _cliCheckNode.default)(`12`);
  await (0, _cliHandleUnhandled.default)();
  (0, _cliWelcome.default)({
    title: "Chakra UI CLI",
    tagLine: `by Chakra UI
${_package.default.description}`,
    bgColor: `#319795`,
    color: `#FFFFFF`,
    bold: true,
    clear: false,
    version: _package.default.version
  });
  (0, _updateNotifier.default)({
    pkg: _package.default,
    shouldNotifyInNpmScript: true,
    updateCheckInterval: 1e3 * 60 * 60 * 24 * 3
  }).notify({
    isGlobal: true
  });
}
